modules/avfilter/filter.c

```diff
@@ -0,0 +1,195 @@
+/**
+ * @file avfilter/filter.c	Video filter using libavfilter -- filtering
+ *
+ * Copyright (C) 2020 Mikhail Kurkov
+ */
+
+#include <string.h>
+#include <libavformat/avformat.h>
+#include <libavfilter/buffersink.h>
+#include <libavfilter/buffersrc.h>
+#include <libavutil/opt.h>
+#include <re.h>
+#include <rem.h>
+#include <baresip.h>
+#include "avfilter.h"
+#include "util.h"
+
+int filter_init(struct avfilter_st *st, char* filter_descr,
+		struct vidframe *frame)
+{
+	char args[512];
+	int err = 0;
+
+	if (!str_isset(filter_descr)) {
+		st->enabled = false;
+		return 0;
+	}
+
+	const AVFilter *buffersrc = avfilter_get_by_name("buffer");
+	const AVFilter *buffersink = avfilter_get_by_name("buffersink");
+	AVFilterInOut *outputs = avfilter_inout_alloc();
+	AVFilterInOut *inputs = avfilter_inout_alloc();
+
+	enum AVPixelFormat src_format = vidfmt_to_avpixfmt(frame->fmt);
+	enum AVPixelFormat pix_fmts[] = { src_format, AV_PIX_FMT_NONE };
+
+	st->filter_graph = avfilter_graph_alloc();
+	st->vframe_in = av_frame_alloc();
+	st->vframe_out = av_frame_alloc();
+	if (!outputs || !inputs || !st->filter_graph ||
+	    !st->vframe_in || !st->vframe_out) {
+		err = AVERROR(ENOMEM);
+		goto end;
+	}
+
+	/* buffer video source */
+	snprintf(args, sizeof(args),
+		 "video_size=%dx%d:pix_fmt=%d:"
+		 "time_base=%d/%d:pixel_aspect=1/1",
+		 frame->size.w, frame->size.h, src_format, 1, 25);
```

Comment:
time_base = 1/25 is hardcoded.

if you want to use the timestamps from the API you can use 1/1000000 as time_base.
(or `VIDEO_TIMEBASE` constant)